tests/spec_test_runners/bls/mod.rs

```diff
@@ -1,47 +1,174 @@
 use crate::test_utils::{load_yaml, TestCase};
 use ethereum_consensus::crypto::{
-    // aggregate, aggregate_verify, fast_aggregate_verify, PublicKey,
-    SecretKey,
-    Signature,
+    aggregate, aggregate_verify, fast_aggregate_verify, PublicKey, SecretKey, Signature,
 };
-// use ethereum_consensus::primitives::Bytes32;
+use ethereum_consensus::primitives::Bytes32;
 use ethereum_consensus::serde as eth_serde;
 use serde::Deserialize;
 use serde_with::{serde_as, DefaultOnError};
 
-pub struct AggregateHandler;
+#[derive(Debug, Deserialize)]
+struct AggregateTestCase {
+    input: Vec<Signature>,
+    output: Option<Signature>,
+}
+
+#[derive(Debug)]
+pub struct AggregateHandler {
+    test_case: AggregateTestCase,
```

Comment:
I'm wondering if we need the extra indirection for `test_case` here

instead we could just inline the `input` and `output` members into the `Handler` types

I will go look at some of the other test types to get a clearer idea, otherwise it just bloats all of the code unnecessarily